src/serializer/ResidualHeapSerializer.js

```diff
@@ -528,7 +529,7 @@ export class ResidualHeapSerializer {
           functionValues,
           val
         );
-        return { body, usedOnlyByResidualFunctions: true };
+        return { body: { type: "DelayInitialization", entries: body }, usedOnlyByResidualFunctions: true };
```

Comment:
I would prefer to have `residualFunctionInitializers` return an `EmitBody` instead of possibly creating multiple `EmitBody` objects for the same `entries` array.